doc/poe/poe_hld.md

```diff
@@ -0,0 +1,370 @@
+- [High-Level Design](#high-level-design)
+  - [PoE flows](#poe-flows)
+    - [PoE data collection](#poe-data-collection)
+    - [PoE configuration](#poe-configuration)
+- [SAI API](#sai-api)
+- [PoE AI](#poe-abstraction-interface)
+- [Configuration and management](#configuration-and-management)
+  - [CLI/YANG model Enhancements](#cli_yang-model-enhancements)
+    - [Show commands](#show-commands)
+    - [Config commands](#config-commands)
+  - [Config DB Enhancements](#config-db-enhancements)
+  - [State DB Enhancements](#state-db-enhancements)
+- [Restrictions/Limitations](#restrictions_limitations)
+
+### Revision
+
+ | Rev | Date |     Author     | Change Description |
+ |:---:|:----:|:--------------:|--------------------|
+ | 0.1 | 03-26-2024 | Volodymyr Mytnyk |  Initial version |
+ | 0.2 | 04-01-2024 | Volodymyr Mytnyk |  Add SWSS cahnges, include LLDP interaction, add example PoE config, mention about warm reboot requirements |
+
+### Scope  
+
+This document provides general information about the PoE (Power over Ethernet) feature implementation in SONiC.
+
+### Definitions/Abbreviations 
+
+|                          |                                |
+|--------------------------|--------------------------------|
+| PoE                      | Power over Ethernet |
+
+### Overview 
+
+Ethernet switch (e.g. campus specific platforms) today often comes with Power over Ethernet (PoE) feature support. PoE functioalility itself is considered as an external functionality from NPU SAI/SDK point of view as it is configurable via external PoE controller.
+
+
+### Requirements
+
+- PoE feature can be enabled via compile option (disabled by default);
+- PoE config persistence during SONiC reboot;
+- Support CLI configurations as mentioned in [config section](#config-commands) and [show section](#show-commands);
+- LLDP support (PoE power reporting only, future releases);
+- Warm boot;
+
+### Architecture Design 
+
+In order to isolate PoE functionality and complexity from existing SONiC design, the following main changes are made:
+
+- Add separate container for PoE control plane routine configuration.
+- Introduce separate SYNCD container to support external PoE Abstraction Interface.
+- Add new PoE tables into database container.
+
+The following diagram explains the changes needed in existing SONiC architecture.
+
+![poe-changes](images/sonic_architecture_with_poe.png)
+
+#### PoE interaction
+
+This section describes the detail interaction of PoE componnents, DB and PoE hardware (controller) in SONiC subsytem.
+
+![poe-interaction](images/poe_interaction.png)
+
+#### SWSS changes
+
+In case to support external and separate PoE SAI, and following SONiC architecture design, SWSS container needs to be extended
+with PoE orch deamon that will handle all PoE ASIC DB requests and does PoE initialization.
+
+![swss-changes](images/poe_swss_poesyncd.png)
+
+#### PoE SYNCD
+
+New PoE SYNCD container componnents:
```

Comment:
what is the reason to have a poe syncd? why normal syncd won't work. isn't that poe feature is captured with SAI api, then normal syncd can support poe as long as the sai library support poe?

Reply:
I believe the motivation is to allow use of an additional instance of SAI. one for NPU and one for PoE. This is assuming that NPU and PoE HW and drivers are developed by different vendors